time-series/time-series-api/src/main/java/com/powsybl/timeseries/ast/UnaryOperation.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.timeseries.ast;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.powsybl.timeseries.TimeSeriesException;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class UnaryOperation implements NodeCalc {
+
+    static final String NAME = "unaryOp";
+
+    public enum Operator {
+        ABS("abs"),
+        NEGATIVE("negative"),
+        POSITIVE("positive");
+
+        Operator(String str) {
+            this.str = Objects.requireNonNull(str);
+        }
+
+        @Override
+        public String toString() {
+            return str;
+        }
+
+        private final String str;
+    }
+
+    public static UnaryOperation abs(NodeCalc child) {
+        return new UnaryOperation(child, Operator.ABS);
+    }
+
+    public static UnaryOperation negative(NodeCalc child) {
+        return new UnaryOperation(child, Operator.NEGATIVE);
+    }
+
+    public static UnaryOperation positive(NodeCalc child) {
+        return new UnaryOperation(child, Operator.POSITIVE);
+    }
+
+    private NodeCalc child;
+
+    private final Operator operator;
+
+    UnaryOperation(NodeCalc child, Operator operator) {
+        this.child = Objects.requireNonNull(child);
+        this.operator = Objects.requireNonNull(operator);
+    }
+
+    public Operator getOperator() {
+        return operator;
+    }
+
+    public NodeCalc getChild() {
+        return child;
+    }
+
+    public void setChild(NodeCalc child) {
+        this.child = child;
```

Comment:
Assert that child is not null

Reply:
Done also added AbstractSingleChildNodeCalc to share code with TimeNodeCalc